crates/telio-proxy/src/proxy.rs

```diff
@@ -283,27 +339,45 @@ impl StateEgress {
     }
 
     async fn send_outbound_data(&mut self, pk: PublicKey, msg: DataMsg) {
-        // if let Some((pk, msg)) = self.input.recv().await {
-        if let (Some((socket, None)), Some(wg_addr)) = (self.sockets.get(&pk), self.wg_addr) {
-            match socket.send_to(msg.get_payload(), wg_addr).await {
-                Ok(_) => {
-                    if self.conn_state.is_err() {
-                        self.conn_state = Ok(());
-                        telio_log_info!("Outbound conn telio -> WG working!");
+        if let Err(err) = self.send_data(pk, msg.clone()).await {
+            let mut should_resend = self.handle_error(err, Some(pk)).await.should_resend();
```

Comment:
So this is still pretty bad, if one socket errors out, assuming others
would be ok, you would be blocked in resend loop.

sleep should happen on wait loops, so update could interup it.

Reply:
This would only be ok if each peer would have an individual task.
So the check should probably be more like "can I send to it now"
if not you skip send to this peer. and move out to upper select.